cmd/exit_sign.go

```diff
@@ -100,110 +100,124 @@ func runSignPartialExit(ctx context.Context, config exitConfig) error {
 		return errors.Wrap(err, "could not match local validator key shares with their counterparty in cluster lock")
 	}
 
-	validator := core.PubKey(config.ValidatorPubkey)
+	shareIdx, err := keystore.ShareIdxForCluster(cl, *identityKey.PubKey())
+	if err != nil {
+		return errors.Wrap(err, "could not determine operator index from cluster lock for supplied identity key")
+	}
 
-	valEth2, err := validator.ToETH2()
+	oAPI, err := obolapi.New(config.PublishAddress, obolapi.WithTimeout(config.PublishTimeout))
 	if err != nil {
-		if (strings.TrimSpace(config.ValidatorPubkey) != "" && !config.ValidatorIndexPresent) || config.ExpertMode {
-			return errors.Wrap(err, "cannot convert validator pubkey to bytes")
-		}
+		return errors.Wrap(err, "could not create obol api client")
 	}
 
-	switch {
-	case config.ExpertMode:
-		ctx = log.WithCtx(ctx, z.U64("validator_index", config.ValidatorIndex), z.Str("validator", validator.String()))
-	case config.ValidatorIndexPresent && !config.ExpertMode:
-		ctx = log.WithCtx(ctx, z.U64("validator_index", config.ValidatorIndex))
-	default:
-		ctx = log.WithCtx(ctx, z.Str("validator", validator.String()))
+	eth2Cl, err := eth2Client(ctx, config.BeaconNodeEndpoints, config.BeaconNodeTimeout, [4]byte(cl.GetForkVersion()))
+	if err != nil {
+		return errors.Wrap(err, "cannot create eth2 client for specified beacon node")
 	}
 
-	shareIdx, err := keystore.ShareIdxForCluster(cl, *identityKey.PubKey())
+	logFields := []z.Field{}
+	if config.ValidatorIndexPresent {
+		logFields = append(logFields, z.U64("validator_index", config.ValidatorIndex))
+	}
+	if config.ValidatorPubkey != "" {
+		logFields = append(logFields, z.Str("validator_pubkey", config.ValidatorPubkey))
+	}
+	ctx = log.WithCtx(ctx, logFields...)
+
+	if config.SkipBeaconNodeCheck {
```

Comment:
Is this conditional the only instance of usage of this flag?

Reply:
Yes, I'm in favour of totally removing it though.